charts/apisix/templates/configmap.yaml

```diff
@@ -28,269 +28,540 @@ data:
       {{- include "apisix.tplvalues.render" (dict "value" $value "context" $) | nindent 6 }}
     {{- end }}
 {{- else }}
-  config.yaml: |-
-    #
-    # Licensed to the Apache Software Foundation (ASF) under one or more
-    # contributor license agreements.  See the NOTICE file distributed with
-    # this work for additional information regarding copyright ownership.
-    # The ASF licenses this file to You under the Apache License, Version 2.0
-    # (the "License"); you may not use this file except in compliance with
-    # the License.  You may obtain a copy of the License at
-    #
-    #     http://www.apache.org/licenses/LICENSE-2.0
-    #
-    # Unless required by applicable law or agreed to in writing, software
-    # distributed under the License is distributed on an "AS IS" BASIS,
-    # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-    # See the License for the specific language governing permissions and
-    # limitations under the License.
-    #
-    apisix:
-      node_listen: {{ .Values.gateway.http.containerPort }}             # APISIX listening port
-      enable_heartbeat: true
-      enable_admin: {{ .Values.admin.enabled }}
-      enable_admin_cors: {{ .Values.admin.cors }}
-      enable_debug: false
-      {{- if or .Values.customPlugins.enabled .Values.apisix.luaModuleHook.enabled }}
-      extra_lua_path: {{ .Values.customPlugins.luaPath }};{{ .Values.apisix.luaModuleHook.luaPath }}
-      {{- end }}
+  {{- if semverCompare ">=2.99.0" .Chart.AppVersion }}
```

Comment:
I think we shouldn't use AppVersion. 

This will make it always use the v3 version of the configuration file, even if the user modifies the image's tag

Reply:
So which variable should we use?

Reply:
https://github.com/apache/apisix-helm-chart/blob/b0fd2cf502699a6d6faa84393cd2239d2f759b68/charts/apisix/values.yaml#L72

maybe just use `tag` for this one.

Reply:
fix